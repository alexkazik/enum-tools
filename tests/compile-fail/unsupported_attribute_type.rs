#![no_implicit_prelude]

use ::enum_tools::EnumTools;

#[derive(EnumTools, Debug, Clone, Copy, Eq, PartialEq)]
#[enum_tools(literal = "x", sorted(y=(z)))]
#[enum_tools = "bad"]
#[repr(i8)]
pub(crate) enum EG {
    #[enum_tools(rename = "A*")]
    A,
    #[enum_tools = "rename"]
    B,
    #[enum_tools(other = "A*")]
    C,
    #[enum_tools(rename(y=z))]
    D,
}

fn main() {}
