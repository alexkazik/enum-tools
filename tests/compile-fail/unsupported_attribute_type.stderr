error: unsupported attribute type
 --> tests/compile-fail/unsupported_attribute_type.rs:6:14
  |
6 | #[enum_tools(literal = "x", sorted(y=(z)))]
  |              ^^^^^^^^^^^^^

error: unsupported attribute type
 --> tests/compile-fail/unsupported_attribute_type.rs:6:36
  |
6 | #[enum_tools(literal = "x", sorted(y=(z)))]
  |                                    ^^^^^

error: unsupported attribute type
 --> tests/compile-fail/unsupported_attribute_type.rs:7:3
  |
7 | #[enum_tools = "bad"]
  |   ^^^^^^^^^^^^^^^^^^

error: unsupported attribute type
  --> tests/compile-fail/unsupported_attribute_type.rs:12:5
   |
12 |     #[enum_tools = "rename"]
   |     ^^^^^^^^^^^^^^^^^^^^^^^^

error: unsupported attribute type
  --> tests/compile-fail/unsupported_attribute_type.rs:14:5
   |
14 |     #[enum_tools(other = "A*")]
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: unsupported attribute type
  --> tests/compile-fail/unsupported_attribute_type.rs:16:5
   |
16 |     #[enum_tools(rename(y=z))]
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^
